diff --git a/configure.in b/configure.in
index 4d87a3f..36d1fc1 100644
--- a/configure.in
+++ b/configure.in
@@ -295,16 +295,16 @@ AM_CONDITIONAL(MISSING_SCANDIR, false)
 
 dnl AC_REPLACE_FUNCS(strlcpy)
 
-AC_CHECK_FUNCS([strlcpy], , [AC_LIBOBJ(missing/strlcpy) AM_CONDITIONAL(MISSING_STRLCPY, true)])
-AC_CHECK_FUNCS([strlcat], , [AC_LIBOBJ(missing/strlcat) AM_CONDITIONAL(MISSING_STRLCAT, true)])
-AC_CHECK_FUNCS([strsep], , [AC_LIBOBJ(missing/strsep) AM_CONDITIONAL(MISSING_STRSEP, true)])
-AC_CHECK_FUNCS([memmem], , [AC_LIBOBJ(missing/memmem) AM_CONDITIONAL(MISSING_MEMMEM, true)])
-AC_CHECK_FUNCS([memcmp], , [AC_LIBOBJ(missing/memcmp) AM_CONDITIONAL(MISSING_MEMCMP, true)])
-AC_CHECK_FUNCS([basename], , [AC_LIBOBJ(missing/basename) AM_CONDITIONAL(MISSING_BASENAME, true)])
-AC_CHECK_FUNCS([getopt_long], , [AC_LIBOBJ(missing/getopt) AM_CONDITIONAL(MISSING_GETOPT_LONG, true)])
-AC_CHECK_FUNCS([strcasestr], , [AC_LIBOBJ(missing/strcasestr) AM_CONDITIONAL(MISSING_STRCASESTR, true)])
-AC_CHECK_FUNCS([scandir], , [AC_LIBOBJ(missing/scandir) AM_CONDITIONAL(MISSING_SCANDIR, true)])
-AC_CHECK_FUNCS([inet_aton], , [AC_LIBOBJ(missing/inet_aton) AM_CONDITIONAL(MISSING_INET_ATON, true)])
+AC_CHECK_FUNCS([strlcpy], , [AC_LIBOBJ(strlcpy) AM_CONDITIONAL(MISSING_STRLCPY, true)])
+AC_CHECK_FUNCS([strlcat], , [AC_LIBOBJ(strlcat) AM_CONDITIONAL(MISSING_STRLCAT, true)])
+AC_CHECK_FUNCS([strsep], , [AC_LIBOBJ(strsep) AM_CONDITIONAL(MISSING_STRSEP, true)])
+AC_CHECK_FUNCS([memmem], , [AC_LIBOBJ(memmem) AM_CONDITIONAL(MISSING_MEMMEM, true)])
+AC_CHECK_FUNCS([memcmp], , [AC_LIBOBJ(memcmp) AM_CONDITIONAL(MISSING_MEMCMP, true)])
+AC_CHECK_FUNCS([basename], , [AC_LIBOBJ(basename) AM_CONDITIONAL(MISSING_BASENAME, true)])
+AC_CHECK_FUNCS([getopt_long], , [AC_LIBOBJ(getopt) AM_CONDITIONAL(MISSING_GETOPT_LONG, true)])
+AC_CHECK_FUNCS([strcasestr], , [AC_LIBOBJ(strcasestr) AM_CONDITIONAL(MISSING_STRCASESTR, true)])
+AC_CHECK_FUNCS([scandir], , [AC_LIBOBJ(scandir) AM_CONDITIONAL(MISSING_SCANDIR, true)])
+AC_CHECK_FUNCS([inet_aton], , [AC_LIBOBJ(inet_aton) AM_CONDITIONAL(MISSING_INET_ATON, true)])
 
 AC_CHECK_LIB(resolv, inet_aton,
    AM_CONDITIONAL(INET_ATON_WANTS_RESOLV, true), 
@@ -836,6 +836,7 @@ AC_CONFIG_FILES(  Makefile
                   man/etter.conf.5
                   share/Makefile 
                   src/Makefile 
+                  src/missing/Makefile
                   src/os/Makefile 
                   src/interfaces/Makefile 
                   src/interfaces/daemon/Makefile 
diff --git a/src/Makefile.am b/src/Makefile.am
index 717b2fb..2bc0dcf 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,7 +3,7 @@
 
 include $(top_srcdir)/Makefile.am.common
 
-SUBDIRS = os interfaces
+SUBDIRS = os interfaces missing
 
 bin_PROGRAMS = ettercap
 
@@ -127,7 +127,7 @@ ettercap_SOURCES += dissectors/ec_socks.c \
 
 ettercap_CFLAGS = @EC_CFLAGS@
 
-ettercap_LDADD = @LIBOBJS@ \
+ettercap_LDADD = missing/libmissing.a \
                  os/libec_os.a \
                  interfaces/daemon/libec_daemon.a \
                  interfaces/text/libec_text.a 
diff --git a/src/missing/Makefile.am b/src/missing/Makefile.am
new file mode 100644
index 0000000..594e470
--- /dev/null
+++ b/src/missing/Makefile.am
@@ -0,0 +1,4 @@
+
+noinst_LIBRARIES = libmissing.a
+libmissing_a_SOURCES =
+libmissing_a_LIBADD = $(LIBOBJS) $(ALLOCA)
