diff -urN ettercap-0.5.4.orig/configure.in ettercap-0.5.4/configure.in
--- ettercap-0.5.4.orig/configure.in	Thu Sep  6 17:51:02 2001
+++ ettercap-0.5.4/configure.in	Thu Sep  6 17:51:40 2001
@@ -102,11 +102,7 @@
 
 dnl configuring path for rpm version...
 
-if test "$RPM_BUILD_ROOT" = ""; then
-	echo "#define INSTALL_PATH \"$prefix/share/ettercap/\"" > ./src/include/ec_install_path.h
-else
-	echo "#define INSTALL_PATH \"/usr/local/share/ettercap/\"" > ./src/include/ec_install_path.h
-fi
+echo "#define INSTALL_PATH \"$libdir/ettercap/\"" > ./src/include/ec_install_path.h
 
 
 dnl ========================
