--- ettercap-NG-0.7.3/configure.in~	2010-06-19 15:46:50.853044834 +0200
+++ ettercap-NG-0.7.3/configure.in	2010-06-19 15:49:05.153040338 +0200
@@ -676,7 +676,7 @@
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
         NCURSINC="-I$withval"
-        NCURSLIB="-L$withval -lpanel -lmenu -lform -lncurses"
+        NCURSLIB="-L$withval -lpanel -lmenu -lform -lncurses -ltinfo"
         AC_DEFINE(HAVE_NCURSES, 1)
         AM_CONDITIONAL(NCURSES, true)
         ac_ec_libncurses=$withval
@@ -688,7 +688,7 @@
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
         NCURSINC="-I$withval/include"
-        NCURSLIB="-L$withval/lib -lpanel -lmenu -lform -lncurses"
+        NCURSLIB="-L$withval/lib -lpanel -lmenu -lform -lncurses -ltinfo"
         AC_DEFINE(HAVE_NCURSES, 1)
         AM_CONDITIONAL(NCURSES, true)
         ac_ec_libncurses=$withval
@@ -700,7 +700,7 @@
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
         NCURSINC="-I$withval/include/ncurses"
-        NCURSLIB="-L$withval/lib -lpanel -lmenu -lform -lncurses"
+        NCURSLIB="-L$withval/lib -lpanel -lmenu -lform -lncurses -ltinfo"
         AC_DEFINE(HAVE_NCURSES, 1)
         AM_CONDITIONAL(NCURSES, true)
         ac_ec_libncurses=$withval
@@ -718,7 +718,7 @@
          AC_MSG_RESULT(no)
      else 
          NCURSINC="-I/usr/include/ncurses"
-         NCURSLIB="-lpanel -lmenu -lform -lncurses"
+         NCURSLIB="-lpanel -lmenu -lform -lncurses -ltinfo"
          AC_DEFINE(HAVE_NCURSES, 1)
          AM_CONDITIONAL(NCURSES, true)
          ac_ec_libncurses=default
@@ -730,7 +730,7 @@
                -a -f /usr/include/panel.h; then
          AC_MSG_RESULT(no)
      else 
-         NCURSLIB="-lpanel -lmenu -lform -lncurses"
+         NCURSLIB="-lpanel -lmenu -lform -lncurses -ltinfo"
          AC_DEFINE(HAVE_NCURSES, 1)
          AM_CONDITIONAL(NCURSES, true)
          ac_ec_libncurses=default
