diff -urN ettercap-0.5.4.orig/aclocal.m4 ettercap-0.5.4/aclocal.m4
--- ettercap-0.5.4.orig/aclocal.m4	Thu Jul 19 06:48:09 2001
+++ ettercap-0.5.4/aclocal.m4	Thu Sep  6 18:39:46 2001
@@ -25,8 +25,8 @@
 AC_DEFUN(EC_LINUX_KERNEL,[
 
 	AC_MSG_CHECKING(Linux kernel version)
-	major=`uname -r  | cut -f1 -d"."`
-	minor=`uname -r  | cut -f2 -d"."`
+	major=`awk -v FS='[".] '/UTS_RELEASE/ { print $2; }' < /usr/include/linux/version.h`
+	minor=`awk -v FS='[".] '/UTS_RELEASE/ { print $3; }' < /usr/include/linux/version.h`
 	uname=`uname -r`
 	AC_MSG_RESULT($uname)
 	if test "$major$minor" -lt 20; then
@@ -198,4 +198,4 @@
     AC_MSG_RESULT($ac_cv_lbl_sockaddr_has_sa_len)
     if test $ac_cv_lbl_sockaddr_has_sa_len = yes ; then
             AC_DEFINE(HAVE_SOCKADDR_SA_LEN,1,[if struct sockaddr has sa_len])
-    fi])
\ No newline at end of file
+    fi])
