--- ettercap-0.8.3/src/interfaces/CMakeLists.txt.orig	2019-06-28 01:19:16.000000000 +0200
+++ ettercap-0.8.3/src/interfaces/CMakeLists.txt	2020-05-23 19:59:18.055324887 +0200
@@ -109,5 +109,5 @@
   VERSION ${VERSION}
   SOVERSION 0
   )
-target_link_libraries(ec_interfaces ${EC_INTERFACES_LIBS} ${EC_LIBS})
+target_link_libraries(ec_interfaces ${EC_INTERFACES_LIBS} lib_ettercap ${EC_LIBS})
 install(TARGETS ec_interfaces DESTINATION ${INSTALL_LIBDIR})
--- ettercap-0.8.3/src/CMakeLists.txt.orig	2019-06-28 01:19:16.000000000 +0200
+++ ettercap-0.8.3/src/CMakeLists.txt	2020-05-23 20:10:25.768374241 +0200
@@ -190,7 +190,7 @@
 
 add_library(lib_ettercap SHARED ${EC_SRC})
 add_dependencies(lib_ettercap libnet)
-target_link_libraries(lib_ettercap ec_interfaces ${EC_LIBS})
+target_link_libraries(lib_ettercap ${EC_LIBS})
 
 if(NOT LIBRARY_BUILD)
   add_subdirectory(interfaces)
@@ -199,7 +199,7 @@
     include_directories(interfaces/curses interfaces/curses/widgets)
   endif()
   add_executable(ettercap ec_parser.c ec_main.c)
-  target_link_libraries(ettercap lib_ettercap ${EC_LIBS})
+  target_link_libraries(ettercap ec_interfaces lib_ettercap ${EC_LIBS})
   if(NOT DISABLE_RPATH)
     set_target_properties(ettercap PROPERTIES
     INSTALL_RPATH ${INSTALL_LIBDIR})
--- ettercap-0.8.3/utils/CMakeLists.txt.orig	2019-06-28 01:19:17.000000000 +0200
+++ ettercap-0.8.3/utils/CMakeLists.txt	2020-05-23 20:23:46.770701506 +0200
@@ -20,7 +20,7 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_executable(etterfilter ${EF_SRC} ${FLEX_EF_SCANNER_OUTPUTS} ${BISON_EF_PARSER_OUTPUTS})
-target_link_libraries(etterfilter lib_ettercap)
+target_link_libraries(etterfilter ec_interfaces lib_ettercap)
 if(NOT DISABLE_RPATH)
   set_target_properties(etterfilter PROPERTIES INSTALL_RPATH ${INSTALL_LIBDIR})
 endif()
@@ -42,7 +42,7 @@
 )
 
 add_executable(etterlog ${EL_SRC})
-target_link_libraries(etterlog lib_ettercap)
+target_link_libraries(etterlog ec_interfaces lib_ettercap)
 
 if(NOT DISABLE_RPATH)
   set_target_properties(etterlog PROPERTIES INSTALL_RPATH ${INSTALL_LIBDIR})
--- ettercap-0.8.3/cmake/Modules/EttercapLibCheck.cmake.orig	2019-06-28 01:19:16.000000000 +0200
+++ ettercap-0.8.3/cmake/Modules/EttercapLibCheck.cmake	2020-05-23 21:00:24.582128273 +0200
@@ -189,7 +189,7 @@
     check_library_exists(bsd strlcat "bsd/string.h" HAVE_STRLCAT)
     check_library_exists(bsd strlcpy "bsd/string.h" HAVE_STRLCPY)
     if(HAVE_STRLCAT OR HAVE_STRLCPY)
-        set(EC_INTERFACES_LIBS ${EC_INTERFACES_LIBS} bsd)
+        set(EC_LIBS ${EC_LIBS} bsd)
     endif()
 endif()
 
@@ -202,7 +202,7 @@
 
 find_library(HAVE_PCAP pcap)
 if(HAVE_PCAP)
-    set(EC_INTERFACES_LIBS ${EC_INTERFACES_LIBS} ${HAVE_PCAP})
+    set(EC_LIBS ${EC_LIBS} ${HAVE_PCAP})
 else()
     message(FATAL_ERROR "libpcap not found!")
 endif()
